crates/turbopack-ecmascript/src/references/mod.rs

```diff
@@ -1327,7 +1331,14 @@ fn analyze_amd_define_with_deps(
     ));
 }
 
-async fn as_abs_path(path: FileSystemPathVc) -> Result<JsValue> {
+/// Currently used to generate the "root" path to a
+/// __filename/__dirname/import.meta.url reference.
+///
+/// TODO: This should be updated to generate a real system path on the fly
+/// during runtime, so that the generated code is constant between systems but
+/// the runtime evaluation can take into account the project's actual root
```

Comment:
It's also used for the evaluation value of a path, and the Todo doesn't apply for that. Maybe split it into two different methods

Reply:
Ah, you're right. Split.